src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AFSamplers.java

```diff
@@ -0,0 +1,112 @@
+        }
+
+        @Override
+        protected double currentValue(final AFState state) { return state.meanBias(); }
+    }
+
+    public static final class BiasRateSampler extends AdaptiveMetropolisSampler<AFState, AFData> {
+        private static final double INITIAL_BIAS_RATE_STEP_SIZE = 10;
+        //not strictly needed, but just in case prevents the model from getting trapped in a silly state during burn-in
+        private static final double MAX_BIAS_RATE = 1000;
+
+        public BiasRateSampler() {
+            super(INITIAL_BIAS_RATE_STEP_SIZE, 0, MAX_BIAS_RATE);
+        }
+
+        @Override
+        protected double proposalLogLikelihood(Double proposedBiasRate, AFState state, AFData data) {
+            return AFModel.logLikelihoodWithProposedBiasRate(state, data, proposedBiasRate);
+        }
+
+        @Override
+        protected double currentValue(final AFState state) { return state.biasRate(); }
+    }
+
+    public static final class POutlierSampler extends AdaptiveMetropolisSampler<AFState, AFData> {
+        private static final double INITIAL_OUTLIER_PROBABILITY_STEP_SIZE = 0.01;
+        private static final double MAX_OUTLIER_PROBABILITY = 0.1;
+
+
+        public POutlierSampler() {
+            super(INITIAL_OUTLIER_PROBABILITY_STEP_SIZE, 0, MAX_OUTLIER_PROBABILITY);
+        }
+
+        @Override
+        protected double proposalLogLikelihood(Double proposedPOutlier, AFState state, AFData data) {
+            return AFModel.logLikelihoodWithProposedPOutlier(state, data, proposedPOutlier);
+        }
+
+        @Override
+        protected double currentValue(final AFState state) { return state.pOutlier(); }
+    }
+
+    // to sample the minor fraction we need an AdaptiveMetropolisSampler that knows which segment it samples
+    public static final class perSegmentMinorFractionSampler extends AdaptiveMetropolisSampler<AFState, AFData> {
+        private static final double INITIAL_MINOR_FRACTION_STEP_SIZE = 0.05;
+        private static final double MAX_MINOR_FRACTION = 0.5;   //by definition!
+
+        private int segment;
+
+        public perSegmentMinorFractionSampler(final int segment) {
+            super(INITIAL_MINOR_FRACTION_STEP_SIZE, 0, MAX_MINOR_FRACTION);
+            this.segment = segment;
+        }
+
+        @Override
+        protected double proposalLogLikelihood(Double proposedMinorFraction, AFState state, AFData data) {
+            final AFParameters parameters = state.getParameters(segment);
+            parameters.setMinorFraction(proposedMinorFraction);
+            return AFModel.segmentLogLikelihood(parameters, data.countsInSegment(segment));
+        }
+
+        @Override
+        protected double currentValue(final AFState state) { return state.minorFractionInSegment(segment); }
+    }
+
+    public static final class MinorFractionsSampler implements Sampler<AFState.MinorFractions, AFState, AFData> {
```

Comment:
`AFState.MinorFractions` is essentially just a typedef for `List<Double>`, the data type of the set of minor fractions over all segments.